sky/spot/recovery_strategy.py

```diff
@@ -24,15 +25,52 @@
 SPOT_DEFAULT_STRATEGY = None
 
 
+def _maybe_retry_until_up(method):
```

Comment:
How about have this take `RETRY_INIT_GAP_SECONDS, retry_until_up` as args, rather than implicitly requiring `self` to have these attributes?